.github/workflows/openapi-lint.yml

```diff
@@ -0,0 +1,39 @@
+name: OpenAPI lint
+
+on:
+  workflow_call:
+
+jobs:
+  openapi-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 19
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Spin up localstack and db
+        uses: yu-ichiro/spin-up-docker-compose-action@v1
+        with:
+          file: docker-compose-local.yml
+
+      - name: Install packages
+        run: npm ci
```

Comment:
Does the generate step need the npm dependency, if not, I'd be tempted to put it down a bit closer to the test step

Reply:
Will do